### PR TITLE
Adapt auto-provisioning scripts to CLI v2, Improve various md files

### PR DESCRIPTION
**Pull request by thomasp-ms**
## Purpose
- Update the yml for the auto provisioning job test to v2. Also added data files, since the previous method of creating dataset from public blob does not work anymore (need to fix it later).
- Improve various md files (mostly the form)

## Does this introduce a breaking change?
```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Can be tested by running the scripts described [here](https://github.com/Azure-Samples/azure-ml-federated-learning/blob/main/automated_provisioning/README.md). (I've done it already)

### CODE REVIEW

File: automated_provisioning/README.md [hunk 1]
```diff
@@ -124,6 +122,7 @@ The command should look something like the below (with the parameters replaced b
 ## Future work
 - Add more validation on input strings if issues come up.
```
Comment (dkmiller) [inline automated_provisioning/README.md:123]: Suggestion: capture the existing and future work as issues on the repo. You can just "lift and shift" (line &mapsto; issue title; empty body).

File: automated_provisioning/sample_job/RunSampleJob.ps1 [hunk 2]
```diff
@@ -18,17 +18,18 @@ az account set --subscription $SubscriptionId
...
-$ExistingMNISTDataset = az ml dataset list --resource-group $ResourceGroup --workspace-name $WorkspaceName --name $DatasetName
-$PublicURL = "https://azureopendatastorage.blob.core.windows.net/mnist/*.gz"
-$PublicURLWithPrefix = "file:" + $PublicURL
-if ($ExistingMNISTDataset.Length -eq 0){
-    Write-Output "The dataset '$DatasetName' does not exist. Creating it from the public URL '$PublicURL'..."
-    az ml dataset create --resource-group $ResourceGroup --workspace-name $WorkspaceName --name $DatasetName --paths $PublicURLWithPrefix
-} else {
-    Write-Output "The dataset '$DatasetName' already exists."
-}
+# REMOVING THIS PART SINCE THE PUBLIC BLOB STORAGE DOES NOT SEEM TO BE AVAILABLE ANYMORE
```
Comment (dkmiller) [inline automated_provisioning/sample_job/RunSampleJob.ps1:21]: Git is your friend. Don't just comment out code, remove it for real.

File: automated_provisioning/sample_job/job.yml [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
+code: ./src
+command: python train.py --data-folder ./mnist-data --regularization 0.8 # ${{ inputs.mnist }}
 environment: azureml:AzureML-sklearn-0.24-ubuntu18.04-py37-cpu:24
-compute: azureml:cont-01-compute # adjust the compute name accordingly
-inputs:
-  mnist:
-    data: azureml:mnist_test_for_fl:1
\ No newline at end of file
+compute: azureml:med-01-compute # adjust the compute name accordingly
+# REMOVING THIS PART SINCE THE PUBLIC BLOB STORAGE DOES NOT SEEM TO BE AVAILABLE ANYMORE
```
Comment (dkmiller) [inline automated_provisioning/sample_job/job.yml:7]: Ditto. Don't just comment out, remove for real.

**Review by dkmiller — CHANGES_REQUESTED** [review]

File: automated_provisioning/README.md (hunk 1 above)
Comment (thomasp-ms) [reply automated_provisioning/README.md:123]: good call, done

**Review by dkmiller — APPROVED** [review]

**Pull request merged**